shared/js/clientApi.js

```diff
@@ -146,7 +146,7 @@ var base = {
    */
   put: function(id, value, filename, callback) {
     $.ajax({
-      url: this.api_base_url + "/" + id + "/" + filename,
+      url: this.api_base_url + "/" + id + (filename ? "/" + filename : ""),
```

Comment:
Not sure if it might be cleaner to have a separate put function for this? Something like put_abuse doesn't really feel like it belongs in the generic clientApi though.